cpp/cmake/thirdparty/get_thrust.cmake

```diff
@@ -40,26 +47,28 @@ function(find_and_configure_thrust VERSION)
         thrust_create_target(cudf::Thrust FROM_OPTIONS)
     endif()
 
-    include(GNUInstallDirs)
-    install(DIRECTORY "${Thrust_SOURCE_DIR}/thrust"
-        DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/libcudf/Thrust/"
-        FILES_MATCHING
-            PATTERN "*.h"
-            PATTERN "*.inl")
-    install(DIRECTORY "${Thrust_SOURCE_DIR}/dependencies/cub/cub"
-        DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/libcudf/Thrust/dependencies/"
-        FILES_MATCHING
-            PATTERN "*.cuh")
+    if(Thrust_SOURCE_DIR) # only install thrust when we have an in-source version
```

Comment:
Forgive me for being ignorant, but shouldn't all this stuff be encapsulated in Thrust's cmake target?

Reply:
Since cudf patches Thrust and therefore it isn't identical to the upstream version we wanted to ensure that won't be used by other projects accidentally. To do this we install Thrust inside of cudfs include directory.

Once we upstream the cudf patches of Thrust we can remove all of this

Reply:
I know Thrust has some machinery for [injecting a custom namespace](https://github.com/NVIDIA/thrust/pull/1464) to ensure our custom thrust doesn't conflict with other versions. Do you think that's relevant here?

Reply:
Namespacing would solve an issue if downstream consumers of cudf need to also bring in a non-patched version of Thrust into a library calling cudf. I/we should sync with the thrust team, it might be best for cudf to always namespace our thrust inclusion going forward.

I would prefer to not use this PR to hash out the larger cudf/thrust issue as this resolves our currently broken CI pipelines.